ledger/distribution/src/main.rs

```diff
@@ -105,38 +104,41 @@ impl S3BlockWriter {
     }
 
     fn write_bytes_to_s3(&self, path: &str, filename: &str, value: &[u8]) {
-        let result: Result<
-            retry::OperationResult<(), ()>,
-            retry::Error<retry::OperationResult<(), RusotoError<PutObjectError>>>,
-        > = retry::retry(
-            retry::delay::Exponential::from_millis(10).map(retry::delay::jitter),
+        use futures::executor::block_on;
+        use retry::{delay, retry, Error, OperationResult};
+        use rusoto_core::RusotoError;
+        use rusoto_s3::{PutObjectError, PutObjectOutput, PutObjectRequest, S3};
+
+        let result: Result<PutObjectOutput, Error<RusotoError<PutObjectError>>> = retry(
+            delay::Exponential::from_millis(10)
+                .map(delay::jitter)
+                .take(10),
```

Comment:
This means we will stop after 10 retries, so the comment on line 140 is wrong. I believe we want to try indefinitely.

Reply:
Updated the comment. I'm opting to make it fail more loudly rather than getting stuck in a retry loop forever.

Reply:
In practice, there have been several issues where a third party was not monitoring this app, and we only noticed when we failed to sync from their node, so the "loud failure" should be spamming error logs with an infinite retry, not dying.